src/joy_api.c

```diff
@@ -109,75 +109,137 @@ static unsigned int joy_splt_format_data(flow_record_t *rec,
                                          unsigned char *data)
 {
     unsigned int i = 0;
+    unsigned int entry_counter = 0;
     unsigned int entries_used = 0;
     unsigned int num_of_pkts = 0;
     unsigned int data_len = 0;
+    uint16_t last_len = 0;
+    int16_t run_count = 0;
     struct timeval ts;
     uint16_t *formatted_data = (uint16_t*)data;
 
-    /* see how many packets we have to process - max is MAX_NFV9_SPLT_SALT_PKTS */
-    num_of_pkts = (rec->op < MAX_NFV9_SPLT_SALT_PKTS) ? rec->op : MAX_NFV9_SPLT_SALT_PKTS;
+    /* figure out actual entries needed
+     * we max out at 10 entries for a SPLT record.
+     */
+    for (i=0; i < rec->op; ++i) {
+        if (rec->pkt_len[i] != last_len ) {
+            /* run has ended increment entry count */
+            if (run_count < 0) {
+                ++entry_counter;
+            }
+
+            /* store last length */
+            last_len = rec->pkt_len[i];
+
+            /* increment entry count, reset run */
+            ++entry_counter;
```

Comment:
Is entry_counter supposed to get incremented twice when run has ended?

Reply:
yes, the first counter gets the run count and the second counter gets the new value